### PR TITLE
deploy docs based on master changes

### PR DESCRIPTION
**Pull request by MateusAndrade**
# Summary
Adding circleci deploy to docs. 

## Test Plan
Since with #2293 now we got Docusaurus set with the docs of the project. I deployed a first version this week manually. 

This PR apply a simple config to circleci step to deploy a new version of the [docs](https://react-native-community.github.io/react-native-camera/) when a PR ir merge on master.

Since I am a noob, with settings related to circleci, etc. It would be interesting someone with more knowledge about this to validate the config I made.

Btw, i hope it helps. :smile: 

### What's required for testing (prerequisites)?
Check if this PR doesn't break the semantic release process, and the build jobs.

## Checklist
- [x] Validate CI config local
- [x]  Apply a change on the docs, to ensure that the ci is working
- [x]  Check if the CI is running the `build` without problems

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -46,3 +46,26 @@ jobs:
...
+            git config --global user.name "<YOUR_NAME>"
+            echo "machine github.com login <GITHUB_USERNAME> password $GH_TOKEN" > ~/.netrc
+            cd website && yarn install && GIT_USER=<GIT_USER> yarn run publish-gh-pages
+
+workflows:
+  version: 2
+  build_and_deploy_docs:
+    jobs:
+      - build
+      - deploy-docs
```
Comment (ronal2do) [inline .circleci/config.yml]: * Cannot find a job named `deploy-docs filters branches only - master` to run in the `jobs:` section of your configuration file.
If you expected a workflow to run, check your config contains a top-level key called 'workflows:'

**Comment by MateusAndrade** [comment]
> You need to create the Circle-CI job

Do you mean in the circleci panel?

**Comment by MateusAndrade** [comment]
@ronal2do i added a filter to run the docs deploy ONLY when a commit is merge on the master. Also, i set some ENVS on CI to properly deploy the docs.

Btw, i must apologize for the commits testing the config. :cry:

**Pull request merged**